src/StoreApi/Authentication.php

```diff
@@ -55,4 +104,84 @@ protected function is_request_to_store_api() {
 		}
 		return 0 === strpos( $GLOBALS['wp']->query_vars['rest_route'], '/wc/store/' );
 	}
+
+	/**
+	 * Get current user IP Address.
+	 *
+	 * X_REAL_IP and CLIENT_IP are custom implementations designed to facilitate obtaining a user's ip through proxies, load balancers etc.
+	 *
+	 * _FORWARDED_FOR (XFF) request header is a de-facto standard header for identifying the originating IP address of a client connecting to a web server through a proxy server.
+	 * Note for X_FORWARDED_FOR, Proxy servers can send through this header like this: X-Forwarded-For: client1, proxy1, proxy2.
+	 * Make sure we always only send through the first IP in the list which should always be the client IP.
+	 * Documentation at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
+	 *
+	 * Forwarded request header contains information that may be added by reverse proxy servers (load balancers, CDNs, and so on).
+	 * Documentation at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+	 * Full RFC at https://datatracker.ietf.org/doc/html/rfc7239
+	 *
+	 * @param boolean $proxy_support Enables/disables proxy support.
+	 * @return string|false
+	 */
+	protected static function get_ip_address( $proxy_support = false ) {
+
+		if ( ! $proxy_support ) {
```

Comment:
Question; Why is proxy support optional? Is there something about it that may break in certain environments? Just wondering why it wouldn't be on by default.

Reply:
If proxy support is ON by default, activating Rate Limiting without being behind another machine will leave this vulnerable to IP spoofing. An attacker could just make requests providing different values for IP addresses carrying headers.